edx_proctoring/views.py

```diff
@@ -489,6 +494,65 @@ def get(self, request):
         elif relevant_attempt:
             data['onboarding_status'] = relevant_attempt.status
 
+        if waffle.switch_is_active(ONBOARDING_STATUS_API):
+            attempt_data = None
+            try:
+                attempt_data = backend.get_onboarding_attempts(course_id=course_id, user_id=user.id)
+            except BackendProviderOnboardingStatusesException as exc:
+                log_message = (
+                    'Failed to use backend onboarding status API endpoint for user_id={user_id}'
+                    'in course_id={course_id} because backend failed to respond. '
+                    'Status: {status}, Response: {response}'.format(
+                        user_id=user.id,
+                        course_id=course_id,
+                        response=str(exc),
+                        status=exc.http_status,
+                    )
+                )
+                LOG.warning(log_message)
+                return Response(data)
+
+            if attempt_data is None:
+                return Response(
```

Comment:
Instead of returning a 404, would it be better to return the data from our logic (if it exists)?

Reply:
while it pains me to remove code we've all put effort into I'm of the opinion that we don't need to maintain two code paths for this once the waffle flag is removed.

Reply:
Make sense. No onboarding status API means no onboarding status from this point on.